lib/driver.js

```diff
@@ -172,18 +172,12 @@ class XCUITestDriver extends BaseDriver {
 
     if (this.opts.browserName) {
       log.info('Safari test requested');
-      this.safari = true;      
+      this.safari = true;
       this.opts.app = undefined;
       this.opts.processArguments = this.opts.processArguments || {};
-      if (this.isRealDevice()) {
-        this.opts.bundleId = SAFARI_LAUNCHER_BUNDLE;
-        this._currentUrl = this.opts.safariInitialUrl || `http://appium.io`;
-        this.opts.processArguments.args = [this._currentUrl];
-      } else {
-        this.opts.bundleId = SAFARI_BUNDLE_ID;
-        this._currentUrl = this.opts.safariInitialUrl || `http://${this.opts.address}:${this.opts.port}/welcome`;
-        this.opts.processArguments.args = ['-u', this._currentUrl];
-      }
+      this.opts.bundleId = SAFARI_BUNDLE_ID;
+      this._currentUrl = this.opts.safariInitialUrl || DEFAULT_SAFARI_URL;
```

Comment:
We don't want to default to something served over the internet. This has caused problems in the past.

The address and port come from the server arguments when Appium is started up. They will both have values.